app/javascript/gobierto_visualizations/webapp/visualizations/treeMapNested.vue

```diff
@@ -327,13 +332,14 @@ export default {
           .join('rect')
           .attr('class', 'children')
           .attr('fill', d => {
+            let valueColor
             if (scaleColor) {
               const { depth } = d
-              const valueColor = depth === 1 ? d.color = this.colors(d.data.name) : d.color = this.colors(d.parent.data.name)
-              return valueColor
+              valueColor = depth === 1 ? this.colors(d.data.name) : depth === 2 ? this.colors(d.parent.data.name) : depth === 3 || depth === 4 ? this.colors(d.data.children[0].contractor) : ''
             } else {
-              return '#12365b'
+              valueColor = '#12365b'
             }
+            return valueColor
```

Comment:
No-else-returns and guard clauses https://dev.to/sylwiavargas/5-ways-to-refactor-if-else-statements-in-js-functions-208e

Reply:
Done!